perfkitbenchmarker/providers/ibmcloud/ibmcloud_disk.py

```diff
@@ -0,0 +1,199 @@
+    logging.info('Volume id: %s', self.vol_id)
+
+  def _Delete(self):
+    """Deletes an external block volume."""
+    if self.vol_id is None:
+      logging.info('Volume %s was not created. Skipping deletion.' % self.name)
+      return
+    volcmd = ibm.IbmAPICommand(self)
+    volcmd.flags['volume'] = self.vol_id
+    logging.info('Volume delete, volcmd.flags %s', volcmd.flags)
+    volcmd.DeleteVolume()
+    logging.info('Volume deleted: %s', self.vol_id)
+    self.vol_id = None
+
+  def _Exists(self):
+    if self.vol_id is None:
+      return False
+    return True
+
+  def _FindVdiskIdByName(self, vdisks, name):
+    """Finds vdisk id by name"""
+    for vdisk in vdisks:
+      if vdisk['name'] == name:
+        return vdisk['id']
+    return None
+
+  def Attach(self, vm):
+    """Attaches the disk to a VM.
+
+    Args:
+      vm: instance of the vm to which the disk will be attached.
+    """
+    if self.vol_id is None:
+      raise errors.Error('Cannot attach remote volume %s' % self.name)
+    if vm.vmid is None:
+      msg = 'Cannot attach remote volume %s to non-existing %s VM' % (self.name,
+                                                                      vm.name)
+      raise errors.Error(msg)
+
+    volcmd = ibm.IbmAPICommand(self)
+    volcmd.flags['name'] = self.name
+    volcmd.flags['instanceid'] = vm.vmid
+    volcmd.flags['volume'] = self.vol_id
+    volcmd.flags['delete'] = True
+
+    # wait till volume is PROVISIONED before attach
+    resp = json.loads(volcmd.ShowVolume())
+    status = resp['status']
+    endtime = time.time() + 300
+    while status != AVAILABLE and time.time() < endtime:
+      time.sleep(2)
+      status = json.loads(volcmd.ShowVolume())['status']
+      logging.info('Checking volume status: %s', status)
+
+    if status != AVAILABLE:
+      logging.error('Failed to create a volume')
+      raise errors.Error('IBMCLOUD ERROR: failed to provision a volume.')
+
+    resp = json.loads(volcmd.InstanceCreateVolume())
+    logging.info('Attached volume, volcmd.flags %s', volcmd.flags)
+    volume = resp['id']
+    logging.info('VDISK ID: %s, vdiskname: %s, instanceid: %s', volume, self.name, vm.vmid)
+    self.attached_vdisk_URI = volume
+    self.attached_vm = vm
+    self._WaitForVolumeAttachment(vm)
+    self._GetDeviceFromVDisk(vm)
+
+  @vm_util.Retry(poll_interval=1, max_retries=-1, timeout=300, log_errors=False,
+                 retryable_exceptions=errors.Resource.RetryableCreationError)
+  def _WaitForVolumeAttachment(self, vm):
+    """Waits and checks until the volume status is attached """
+    if self.vol_id is None:
+      return
+    volcmd = ibm.IbmAPICommand(self)
+    volcmd.flags['instanceid'] = self.attached_vm.vmid
+    volcmd.flags['volume'] = self.vol_id
+    logging.info('Checking volume on instance %s, volume: %s', vm.vmid, self.vol_id)
+    resp = volcmd.InstanceShowVolume()
+    logging.info('Show volume resp: %s', resp)
+    status = None
+    if resp:
```

Comment:
I would just do the check inside the loop so you only have the code once, so remove this part and put the time.sleep(2) after you do the check (line 156)

Reply:
agree, just one call used.